src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryCachingCleanup.java

```diff
@@ -49,8 +50,15 @@ private boolean removeIfExpiredCacheDirectory(FilePath library) throws IOExcepti
         final FilePath lastReadFile = new FilePath(library, LibraryCachingConfiguration.LAST_READ_FILE);
         if (lastReadFile.exists()) {
             if (System.currentTimeMillis() - lastReadFile.lastModified() > TimeUnit.DAYS.toMillis(EXPIRE_AFTER_READ_DAYS)) {
-                library.deleteRecursive();
-                library.withSuffix("-name.txt").delete();
+        	ReentrantReadWriteLock retrieveLock = LibraryAdder.getReadWriteLockFor(library.getName());
+                if (retrieveLock.writeLock().tryLock()) {
```

Comment:
I think it is fine to block interruptibly waiting for the lock here:

```suggestion
                if (retrieveLock.writeLock().lockInterruptibly()) {
```